src/lib/utils.ts

```diff
@@ -46,3 +49,6 @@ export const unpackThrowable = <T>(throwable: () => T): ThrowableToTupleResult<T
     return [UnknownError.normalize(maybeError), null];
   }
 };
+
+export const isNotNil = <T>(val: T | null | undefined): val is T =>
```

Comment:
Could we use name `nullish` instead? It's used already by zod: https://zod.dev/?id=nullish. `Nil` feels for me like writing Swift 😄 

```suggestion
export const isNotNullish = <T>(val: T | null | undefined): val is T =>
```